xfel/merging/application/filter/experiment_filter.py

```diff
@@ -61,6 +61,11 @@ def check_cluster(self, experiment):
     features = np.array(features).reshape(1,-1)
     cov=self.cluster_data["populations"].fit_components[self.params.filter.unit_cell.cluster.covariance.component]
     m_distance = sqrt(cov.mahalanobis(features))
+    for other in self.params.filter.unit_cell.cluster.covariance.skip_component:
+      skip_cov = self.cluster_data["populations"].fit_components[other]
+      skip_distance = sqrt(skip_cov.mahalanobis(features))
+      if skip_distance < self.params.filter.unit_cell.cluster.covariance.skip_mahalanobis:
```

Comment:
@dwpaley Correct me if I'm wrong, but the output of this comparison operation in line 67 is a non-empty boolean list, which in Python always evaluates to True, so a `False` will be returned instead of a boolean iterable in all cases. Shouldn't this part rather declare an additional boolean array, to be considered alongside the boolean array in the return statement in line 69?

Reply:
I'm thinking about something along the lines of:

    accept_flag = m_distance < self.params.filter.unit_cell.cluster.covariance.mahalanobis
    dont_skip_flag = skip_distance > self.params.filter.unit_cell.cluster.covariance.skip_mahalanobis
    return accept_flag & dont_skip_flag

Reply:
This whole method `check_cluster` gets called on 1 experiment at a time, meaning `skip_cov.mahalanobis(features)` is just a number (the normalized distance from the current cell to the cluster), not an array. The new "skip" mode uses the same syntax as the `m_distance` filtering on lines 63 and 69.

Reply:
I must have used something similar that acted on an iterable, hence my confusion. Elegant solution 👍